src/citrine/_rest/collection.py

```diff
@@ -222,3 +228,60 @@ def _check_experimental(self, data):
                 "\n  ".join(data.get("experimental_reasons") or ["Unknown reason"])
             )
             warnings.warn(msg)
+
+    def _poll_for_job_completion(self, project_id: UUID,
+                                 job: Union[JobSubmissionResponse, UUID], *,
```

Comment:
```suggestion
    def _poll_for_job_completion(self, project_id: Union[UUID, str],
                                 job: Union[JobSubmissionResponse, UUID, str], *,
```
Matching other signatures around UUIDs in file.